server/cluster/cluster.go

```diff
@@ -1504,6 +1509,13 @@ func (c *RaftCluster) FitRegion(region *core.RegionInfo) *placement.RegionFit {
 	return c.GetRuleManager().FitRegion(c, region)
 }
 
+// GetAntiRuleManager returns the anti rule manager reference.
+func (c *RaftCluster) GetAntiRuleManager() *anti.AntiRuleManager {
+	c.RLock()
+	defer c.RUnlock()
```

Comment:
Though the `c.antiRuleManager` could be concurrently read, the only place to set it is in the `InitCluster`. So I'm wondering whether the lock here is necessary.